addons/web/models/models.py

```diff
@@ -226,51 +228,542 @@ def cleanup(vals: dict) -> dict:
+                overriding the natural sort ordering of the groups,
+                see also :meth:`~.search`.
+        :return: list of dictionaries (one dictionary for each group) containing:
+
+                    * the groupby values: {groupby[i]: <value>}
+                    * the aggregate values: {aggregates[i]: <value>}
+                    * __extra_domain: list of tuples specifying the group search criteria
+                    * __fold: boolean if a fold_name is set on the comodel and read_group_expand is activated
+
+        :rtype: [{'groupy_spec': value, ...}, ...]
+        :raise AccessError: if user is not allowed to access requested information
+        """
+        groupby = tuple(groupby)
+        aggregates = tuple(aggregates)
+
+        if not order:
+            order = ', '.join(groupby)
+
+        groups = self._read_group(
+            domain, groupby,
+            # Avoid recordset in _web_read_group_format as aggregate
+            tuple(agg.replace(':recordset', ':array_agg') for agg in aggregates),
+            having=having, offset=offset, limit=limit, order=order,
+        )
+
+        # Note: group_expand is only done if the limit isn't reached and when the offset == 0
+        # to avoid inconsistency in the web client pager. Anyway, in practice, this feature should
+        # be used only when there are few groups (or without limit for the kanban view).
+        if (
+            not offset and (not limit or len(groups) < limit)
+            and self._web_read_group_field_expand(groupby)
+        ):
+            # It doesn't respect the order with aggregates inside
+            expand_groups = self._web_read_group_expand(domain, groups, groupby[0], aggregates, order)
+            if not limit or len(expand_groups) < limit:
+                # Ditch the result of expand_groups because the limit is reached and to avoid
+                # returning inconsistent result inside length of web_read_group
+                groups = expand_groups
+
+        fill_temporal = self.env.context.get('fill_temporal')
+        if groupby and (fill_temporal or isinstance(fill_temporal, dict)):
+            if limit or offset:
+                raise ValueError('You cannot used fill_temporal with a limit or an offset')
+            if not isinstance(fill_temporal, dict):
+                fill_temporal = {}
+            # This assumes that existing data is sorted by field 'groupby_name'
+            groups = self._web_read_group_fill_temporal(groups, groupby, aggregates, **fill_temporal)
+
+        return self._web_read_group_format(groupby, aggregates, groups)
+
+    def _web_read_group_field_expand(self, groupby):
+        """ Return the field that should be expand """
+        if (
+            len(groupby) == 1
+            and self.env.context.get('read_group_expand')
+            and (field := self._fields[groupby[0].split('.')[0].split(':')[0]])
+            and field.group_expand
+        ):
+            return field
```

Comment:
Return explicit value.
```suggestion
            return field
        return None
```